bulk_indexer.go

```diff
@@ -274,9 +274,9 @@ func (b *BulkIndexer) Flush(ctx context.Context) (BulkIndexerResponseStat, error
 
 	if b.config.MaxDocumentRetries > 0 {
 		if cap(b.copyBuf) < b.buf.Len() {
-			b.copyBuf = slices.Grow(b.copyBuf, b.buf.Len()-cap(b.copyBuf))
-			b.copyBuf = b.copyBuf[:cap(b.copyBuf)]
+			b.copyBuf = slices.Grow(b.copyBuf, b.buf.Len()-len(b.copyBuf))
 		}
+		b.copyBuf = b.copyBuf[:cap(b.copyBuf)]
 		n := copy(b.copyBuf, b.buf.Bytes())
 		b.copyBuf = b.copyBuf[:n]
```

Comment:
Grow does not mean that we only grow the capacity by the given number, it rather ensures we can *at-least* add n elements. So, I don't think we can simplify here.

Reply:
> 		b.copyBuf = b.copyBuf[:len(b.copyBuf)]

Also, this will not work since Grow doesn't change length, and copy works on length. We would have to use `b.buf.Len()`. I don't see a way we can simplify (as in reduce complexity).